core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -129,31 +129,42 @@ private[chisel3] trait HasId extends InstanceId {
     this
   }
 
-  // Private internal version of suggestName that tells you if the name changed
-  // Returns Some(old name, old prefix) if name changed, None otherwise
-  private[chisel3] def _suggestNameCheck(seed: => String): Option[(String, Prefix)] = {
-    val oldSeed = this.seedOpt
-    val oldPrefix = this.naming_prefix
-    suggestName(seed)
-    if (oldSeed.nonEmpty && (oldSeed != this.seedOpt || oldPrefix != this.naming_prefix)) {
-      Some(oldSeed.get -> oldPrefix)
-    } else None
-  }
-
-  /** Takes the first seed suggested. Multiple calls to this function will be ignored.
+  def _suggestNameInternal(seed: => String): this.type = {
+    if (suggested_seed.isEmpty) suggested_seed = Some(seed)
+    naming_prefix = Builder.getPrefix
+    this
+  }
+
+  /** Takes the first seed suggested. Multiple calls to this function will be ignored (and will become an error in future).
     * If the final computed name conflicts with another name, it may get uniquified by appending
     * a digit at the end.
     *
     * Is a higher priority than [[autoSeed]], in that regardless of whether [[autoSeed]]
     * was called, [[suggestName]] will always take precedence.
     *
+    * @note calling this after the name has already been computed will become an error in the future.
+    *
     * @param seed The seed for the name of this component
     * @return this object
     */
   def suggestName(seed: => String): this.type = {
-    if (suggested_seed.isEmpty) suggested_seed = Some(seed)
-    naming_prefix = Builder.getPrefix
-    this
+    require(Builder.hasDynamicContext, s"suggestName (${seed}) should only be called from a Builder context.")
```

Comment:
I think this should also be a deprecation rather than error (for purposes of backporting to 3.5.x). It currently just silently does nothing: https://scastie.scala-lang.org/nfiyNDBaSJWx4GKY4cxxOg

Reply:
hmm, i thought I had a unit test for this error case so I am surprised it silently does nothing...

Reply:
ok I will make it a deprecation error, but still confused about why you say it silently does nothing given the test for it

Reply:
Sorry, by currently I mean in Chisel 3.5.3 it does nothing (see my Scastie). I'm saying it should be a warning in 3.5.4 and an error in 3.6.0.